src/Symfony/Component/Form/Form.php

```diff
@@ -726,11 +726,13 @@ public function isEmpty()
             }
         }
 
-        return FormUtil::isEmpty($this->modelData) ||
-            // arrays, countables
-            ((\is_array($this->modelData) || $this->modelData instanceof \Countable) && 0 === \count($this->modelData)) ||
-            // traversables that are not countable
-            ($this->modelData instanceof \Traversable && 0 === iterator_count($this->modelData));
+        if (!method_exists($this->config, 'getIsEmptyCallback')) {
+            @trigger_error(sprintf('Not implementing the "%s::getIsEmptyCallback()" method in "%s" is deprecated since Symfony 5.1.', FormConfigInterface::class, \get_class($this->config)), E_USER_DEPRECATED);
+
+            return FormConfigBuilder::isEmpty($this->modelData);
```

Comment:
If we move this to the form type config, we could add the option here if we do not want to duplicate the closure:

```php
return $this->config->getOption('is_empty_callback')();
```